packages/core/src/molecule/signed.ts

```diff
@@ -0,0 +1,211 @@
+import { bytesFrom } from "../bytes/index.js";
+import { Num, numFrom, NumLike} from "../num/index.js";
+import { Codec, option, vector } from "./codec.js";
+
+/**
+ * Codec for encoding and decoding 8-bit signed integers.
+ */
+export const Int8 = Codec.from<NumLike, number>({
+  byteLength: 1,
+  /**
+   * Encodes a number-like value into Bytes.
+   * @param numLike - The number-like value to encode.
+   * @returns Bytes containing the encoded value.
+   */
+  encode: (numLike) => {
+    const encoded = new Uint8Array(1);
+    new DataView(encoded.buffer).setInt8(0, Number(numLike));
+    return encoded;
+  },
+  /**
+   * Decodes Bytes into a number.
+   * @param bytesLike - The bytes-like input to decode.
+   * @returns The decoded 8-bit signed integer.
+   */
+  decode: (bytesLike) => {
+    const bytes = bytesFrom(bytesLike);
+    return new DataView(bytes.buffer).getInt8(0);
+  },
+});
+
+/**
+ * Option codec for 8-bit signed integers.
+ */
+export const Int8Opt = option(Int8);
+
+/**
+ * Vector codec for arrays of 8-bit signed integers.
+ */
+export const Int8Vec = vector(Int8);
+
+/**
+ * Creates a codec for encoding and decoding 16-bit signed integers.
+ * @param littleEndian - Indicates whether to use little-endian byte order.
+ * @returns A codec for 16-bit signed integers.
+ */
+function int16Number(littleEndian: boolean): Codec<NumLike, number> {
```

Comment:
I prefer to have sth like `intNumber(byteLength, littleEndian)` to reduce duplicate code, though that means we might not be able to use `DataView`.

`ccc.numToByte` added support for negative numbers. Maybe that helps

https://github.com/ckb-devrel/ccc/pull/177